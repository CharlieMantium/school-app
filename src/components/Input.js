import React from 'react';
import PropTypes from 'prop-types';

const Input = ({ type, placeholder, value, onChange, onBlur, errorMsg }) => (
  <div>
    <input
      type={type}
      placeholder={placeholder}
      value={value}
      onChange={onChange}
      onBlur={onBlur}
    />
    {errorMsg && <p>{errorMsg}</p>}
  </div>
);

Input.defaultProps = {
  type: 'text',
  placeholder: '',
  value: '',
  onChange: () => {},
  onBlur: () => {},
  errorMsg: '',
};

Input.propTypes = {
  type: PropTypes.string,
  placeholder: PropTypes.string,
  value: PropTypes.string,
  onChange: PropTypes.func,
  onBlur: PropTypes.func,
  errorMsg: PropTypes.string,
};

export default Input;
