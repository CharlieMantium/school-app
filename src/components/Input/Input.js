import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { FormattedMessage } from 'react-intl';

import { colors, effects, spacing } from 'styles/base';

const DescriptionsWrapper = styled.div`
  display: flex;
  justify-content: space-between;
`;

const Description = styled.p`
  color: ${props => (props.error ? colors.red : colors.white)};
  text-shadow: ${props => (props.error ? effects.outline(colors.black) : 'none')};
  margin: ${spacing.mSize} ${spacing.sSize} 0 ${spacing.sSize};
`;

const StyledInput = styled.input`
  border: ${spacing.xsSize} solid ${colors.darkGrey};
  border-radius: ${spacing.sSize};
  margin: auto;
  padding: ${spacing.xxsSize} ${spacing.sSize};
  width: 100%;

  &:focus {
    border-color: ${colors.black};
  }

  &:hover {
    border-color: ${colors.green};
  }
`;

const Input = ({ type, placeholder, value, onChange, onBlur, errorMsg }) => (
  <>
    <DescriptionsWrapper>
      <Description>
        <FormattedMessage id={`inputDesc-${placeholder}`} defaultMessage={placeholder} />
      </Description>
      {errorMsg && (
        <Description error data-test="error-message">
          {errorMsg}
        </Description>
      )}
    </DescriptionsWrapper>
    <StyledInput
      type={type}
      placeholder={placeholder}
      value={value}
      onChange={onChange}
      onBlur={onBlur}
      data-test="input"
    />
  </>
);

Input.propTypes = {
  type: PropTypes.string,
  placeholder: PropTypes.string,
  value: PropTypes.string,
  onChange: PropTypes.func.isRequired,
  onBlur: PropTypes.func,
  errorMsg: PropTypes.string,
};

Input.defaultProps = {
  type: 'text',
  placeholder: '',
  value: '',
  onBlur: null,
  errorMsg: '',
};

export default Input;
