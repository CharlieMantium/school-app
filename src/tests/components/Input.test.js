import React from 'react';
import { shallow } from 'enzyme';

import Input from '../../components/Input';
import testData from '../fixtures/inputData';

describe('Input', () => {
  it('should render Input correctly with test data', () => {
    const wrapper = shallow(<Input {...testData} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render Input and error message correctly with errorMsg', () => {
    const wrapper = shallow(<Input {...testData} errorMsg="error message" />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render Input correctly without non-required data', () => {
    const wrapper = shallow(<Input onChange={() => {}} />);
    expect(wrapper).toMatchSnapshot();
  });
});
