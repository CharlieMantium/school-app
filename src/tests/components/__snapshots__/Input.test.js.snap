// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should render Input and error message correctly with errorMsg 1`] = `
<div>
  <input
    onBlur={[Function]}
    onChange={[Function]}
    placeholder="Put your text here."
    type="text"
    value="this input value"
  />
  <p>
    error message
  </p>
</div>
`;

exports[`Input should render Input correctly with test data 1`] = `
<div>
  <input
    onBlur={[Function]}
    onChange={[Function]}
    placeholder="Put your text here."
    type="text"
    value="this input value"
  />
</div>
`;

exports[`Input should render Input correctly without non-required data 1`] = `
<div>
  <input
    onBlur={null}
    onChange={[Function]}
    placeholder=""
    type="text"
    value=""
  />
</div>
`;
